llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -2412,6 +2416,7 @@ void llvm::UpgradeIntrinsicCall(CallBase *CI, Function *NewFn) {
   Builder.SetInsertPoint(CI->getParent(), CI->getIterator());
 
   if (!NewFn) {
+    bool ShouldRemove = true;
```

Comment:
nit: Can we add a short comment explaining this variable?

Also, possibly worth inverting it to "FallthroughToDefaultUpgrade" or something, which is slightly more descriptive. Not 100% sure on that, but if you can think of a way to make it slightly more descriptive that would be good. YMMV, this is ok if not.